src/Microsoft.ApplicationInsights/DataContracts/ExceptionTelemetry.cs

```diff
@@ -286,32 +337,44 @@ private void ConvertExceptionTree(Exception exception, ExceptionDetails parentEx
             }
         }
 
-        private void UpdateExceptions(Exception exception)
+        private void UpdateData(Exception exception, ExceptionInfo exceptionInfo)
         {
-            // collect the set of exceptions detail info from the passed in exception
-            List<ExceptionDetails> exceptions = new List<ExceptionDetails>();
-            this.ConvertExceptionTree(exception, null, exceptions);
-
-            // trim if we have too many, also add a custom exception to let the user know we're trimmed
-            if (exceptions.Count > Constants.MaxExceptionCountToSave)
+            if (this.isCreatedFromExceptionInfo)
             {
-                // TODO: when we localize these messages, we should consider not using InvariantCulture
-                // create our "message" exception.
-                InnerExceptionCountExceededException countExceededException = new InnerExceptionCountExceededException(
-                    string.Format(
-                        CultureInfo.InvariantCulture,
-                        "The number of inner exceptions was {0} which is larger than {1}, the maximum number allowed during transmission. All but the first {1} have been dropped.",
-                        exceptions.Count,
-                        Constants.MaxExceptionCountToSave));
-
-                // remove all but the first N exceptions
-                exceptions.RemoveRange(Constants.MaxExceptionCountToSave, exceptions.Count - Constants.MaxExceptionCountToSave);
-
-                // we'll add our new exception and parent it to the root exception (first one in the list)
-                exceptions.Add(ExceptionConverter.ConvertToExceptionDetails(countExceededException, exceptions[0]));
+                this.Data = exceptionInfo ?? throw new ArgumentNullException(nameof(exceptionInfo));
```

Comment:
>Data [](start = 21, length = 4)

who sanitize the stack frames for this code path?

Reply:
I don't want to sanitize it here. For my particular scenario, it's already sanitized. For a broader scenario where the customer is supplying custom data we don't want it to be changed in a pretty much random way (which is what sanitizing does right now). There is an argument to be made for excessive length, but I think it should be up to the customer to make that determination in this particular scenario.